enricher/api/src/main/java/io/fabric8/maven/enricher/api/visitor/MetadataVisitor.java

```diff
@@ -166,7 +167,20 @@ protected Kind getKind() {
 
         @Override
         protected ObjectMeta getOrCreateMetadata(PodTemplateSpecBuilder item) {
-            return item.hasMetadata() ? item.buildMetadata() : item.withNewMetadata().endMetadata().buildMetadata();
+            if (Boolean.TRUE.equals(item.hasMetadata())) {
+                if (item.buildMetadata().getLabels() == null) {
+                    item.editMetadata().withLabels(Collections.EMPTY_MAP).endMetadata();
```

Comment:
Please, use `Collections.emptyMap()` (returns a generic non-raw type) instead of `Collections.EMPTY_MAP`

Reply:
There are too many duplicate lines in the class, find a proposal to refactor here:

https://github.com/rohanKanojia/fabric8-maven-plugin/pull/4/